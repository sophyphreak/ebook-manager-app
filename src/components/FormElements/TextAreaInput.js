import React, { Component } from 'react';
import {
  FormFeedback,
  FormGroup,
  Label,
  Input
} from 'reactstrap';
import isValid from './isValid/isValid';

const TextAreaInput = ({ 
  hasError,
  errorMessage,
  label, 
  value, 
  onChange 
}) => (
  <FormGroup>
    <Label>{label}</Label>
    <Input
      type="textarea"
      value={value}
      onChange={onChange}
      valid={isValid(hasError, errorMessage)}
    />
    {errorMessage && <FormFeedback>{errorMessage}</FormFeedback>}
  </FormGroup>
);

export default TextAreaInput;