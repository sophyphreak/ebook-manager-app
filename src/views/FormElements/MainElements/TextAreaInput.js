import React from 'react';
import { FormFeedback, FormGroup, Label, Input } from 'reactstrap';
import isValid from '../getValidity/isValid'
import isInvalid from '../getValidity/isInvalid';

const TextAreaInput = ({ hasError, errorMessage, label, value, onChange }) => (
  <FormGroup>
    <Label>{label}</Label>
    <Input
      className="textarea"
      type="textarea"
      value={value}
      onChange={onChange}
      valid={isValid(hasError, errorMessage)}
      invalid={isInvalid(hasError, errorMessage)}
    />
    {errorMessage && <FormFeedback>{errorMessage}</FormFeedback>}
  </FormGroup>
);

export default TextAreaInput;
