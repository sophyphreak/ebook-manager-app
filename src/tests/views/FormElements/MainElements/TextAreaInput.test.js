import React from 'react';
import { shallow } from 'enzyme';
import TextAreaInput from '../../../../views/FormElements/MainElements/TextAreaInput';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<TextAreaInput />);
});

test('should render TextAreaInput correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
