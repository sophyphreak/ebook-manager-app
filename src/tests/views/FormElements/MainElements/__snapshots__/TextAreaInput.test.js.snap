// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextAreaInput correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextAreaInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        />,
        <Input
          className="textarea"
          invalid={false}
          type="textarea"
          valid={false}
        />,
        undefined,
      ],
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "textarea",
          "invalid": false,
          "onChange": undefined,
          "type": "textarea",
          "valid": false,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          />,
          <Input
            className="textarea"
            invalid={false}
            type="textarea"
            valid={false}
          />,
          undefined,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "textarea",
            "invalid": false,
            "onChange": undefined,
            "type": "textarea",
            "valid": false,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
